clients/instance/ibm-pi-virtual-serial-number.go

```diff
@@ -46,10 +46,76 @@ func (f *IBMPIVSNClient) GetAll(pvmInstanceID *string) (models.VirtualSerialNumb
 	}
 	resp, err := f.session.Power.PCloudVirtualSerialNumber.PcloudVirtualserialnumberGetall(params, f.session.AuthInfo(f.cloudInstanceID))
 	if err != nil {
-		return nil, ibmpisession.SDKFailWithAPIError(err, fmt.Errorf("failed to Get all virtual serial numbers in %s :%w", f.cloudInstanceID, err))
+		return nil, ibmpisession.SDKFailWithAPIError(err, fmt.Errorf("failed to get all virtual serial numbers in %s :%w", f.cloudInstanceID, err))
 	}
 	if resp == nil || resp.Payload == nil {
-		return nil, fmt.Errorf("failed to Get all virtual serial numbers in %s", f.cloudInstanceID)
+		return nil, fmt.Errorf("failed to get all virtual serial numbers in %s", f.cloudInstanceID)
 	}
 	return resp.Payload, nil
 }
+
+// Update Virtual Serial Nunber
+func (f *IBMPIVSNClient) Update(id string, body *models.UpdateVirtualSerialNumber) (*models.GetServerVirtualSerialNumber, error) {
+	params := p_cloud_virtual_serial_number.NewPcloudVirtualserialnumberPutParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIUpdateTimeOut).WithVirtualSerialNumber(id).
+		WithBody(body)
+	resp, err := f.session.Power.PCloudVirtualSerialNumber.PcloudVirtualserialnumberPut(params, f.session.AuthInfo(f.cloudInstanceID))
+	if err != nil {
+		return nil, ibmpisession.SDKFailWithAPIError(err, fmt.Errorf("failed to update virtual serial number %s :%w", id, err))
+	}
+	if resp == nil || resp.Payload == nil {
+		return nil, fmt.Errorf("failed to Update virtual serial number %s", id)
+	}
+	return resp.Payload, nil
+}
+
+// Delete Virtual Serial Number
+func (f *IBMPIVSNClient) Delete(id string) error {
+	params := p_cloud_virtual_serial_number.NewPcloudVirtualserialnumberDeleteParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIDeleteTimeOut).
+		WithVirtualSerialNumber(id)
+	_, err := f.session.Power.PCloudVirtualSerialNumber.PcloudVirtualserialnumberDelete(params, f.session.AuthInfo(f.cloudInstanceID))
+	if err != nil {
+		return ibmpisession.SDKFailWithAPIError(err, fmt.Errorf("failed to delete virtual serial number %s :%w", id, err))
+	}
+	return nil
+}
+
+// PVM Instance Delete VSN
+func (f *IBMPIVSNClient) PVMInstanceDeleteVSN(pvmInstanceID string, body *models.DeleteServerVirtualSerialNumber) error {
```

Comment:
In the CLI we use assign and unassign as the wording if you want to keep the wording similar

Reply:
That wording can be kept the same in TF,  it does not matter here.